game/state/shared/aequipment.cpp

```diff
@@ -702,12 +702,19 @@ void AEquipment::fire(GameState &state, Vec3<float> targetPosition, StateRef<Bat
 	auto unit = ownerAgent->unit;
 	auto payload = getPayloadType();
 	Vec3<float> originalTarget = targetPosition;
+	int number_of_shots;
 
 	if (payload->fire_sfx)
 	{
 		fw().soundBackend->playSample(payload->fire_sfx, unit->position);
 	}
 
+	number_of_shots = this->type->burst;
+	if (number_of_shots == 0)
```

Comment:
Similarly, would setting the default to "1" mean less code here?

Reply:
See response to the same code in vequipment.cpp

This was the only way i could get around <burst> not being specified and returning a null value or modders simply giving "0" as a value rather than removing the line completely